src/sagas.js

```diff
@@ -384,6 +390,39 @@ function* selectVmDetail (action) {
   yield fetchSingleVm(getSingleVm({ vmId: action.payload.vmId })) // async data refresh
 }
 
+function* getConsoleOptions (action) {
+  const options = OptionsManager.loadConsoleOptions(action.payload)
+  yield put(setConsoleOptions({ vmId: action.payload.vmId, options }))
+}
+
+function* saveConsoleOptions (action) {
+  OptionsManager.saveConsoleOptions(action.payload)
+  yield getConsoleOptions({ payload: { vmId: action.payload.vmId } })
+}
+
+function* autoConnectCheck (action) {
```

Comment:
Can be the code shared with getConsoleVm() ?
Ideally fire GET_CONSOLE_VM action from autoConnectCheck()